lib/new_relic/agent/instrumentation/sidekiq.rb

```diff
@@ -14,32 +14,44 @@
   end
 
   executes do
-    class NewRelic::SidekiqInstrumentation
-      include NewRelic::Agent::Instrumentation::ControllerInstrumentation
+    module NewRelic::SidekiqInstrumentation
+      class Server
+        include NewRelic::Agent::Instrumentation::ControllerInstrumentation
 
-      # Client middleware has additional parameters, and our tests use the
-      # middleware client-side to work inline.
-      def call(worker, msg, queue, *_)
-        trace_args = if worker.respond_to?(:newrelic_trace_args)
-          worker.newrelic_trace_args(msg, queue)
-        else
-          self.class.default_trace_args(msg)
+        # Client middleware has additional parameters, and our tests use the
+        # middleware client-side to work inline.
+        def call(worker, msg, queue, *_)
+          trace_args = if worker.respond_to?(:newrelic_trace_args)
+            worker.newrelic_trace_args(msg, queue)
+          else
+            self.class.default_trace_args(msg)
+          end
+          trace_info = msg.delete("newrelic")
+          
+          perform_action_with_newrelic_trace(trace_args) do
+            NewRelic::Agent::Transaction.merge_untrusted_agent_attributes(msg['args'], :'job.sidekiq.args',
+              NewRelic::Agent::AttributeFilter::DST_NONE)
+              
+            ::NewRelic::Agent::DistributedTracing.accept_distributed_trace_payload(trace_info) if trace_info
```

Comment:
This API call is deprecated along with it's companion create_distributed_trace_payload.  Intead, use:
```
::NewRelic::Agent::DistributedTracing::accept_distributed_trace_headers(trace_info) if trace_info
```